qiita_pet/handlers/study_handlers/description_handlers.py

```diff
@@ -83,865 +60,81 @@ def _to_int(value):
-        user : User
-            The current user object
-        callback : function
-            The callback function to call with the results once the processing
-            is done
-        """
-        ppd_id = int(self.get_argument('preprocessed_data_id'))
-
-        try:
-            PreprocessedData.delete(ppd_id)
-            msg = ("Preprocessed data %d has been deleted" % ppd_id)
-            msg_level = "success"
-            ppd_id = None
-        except Exception as e:
-            msg = ("Couldn't remove preprocessed data %d: %s" %
-                   (ppd_id, str(e)))
-            msg_level = "danger"
-
-        callback((msg, msg_level, 'preprocessed_data_tab', ppd_id, None))
-
-    def delete_processed_data(self, study, user, callback):
-        """Delete the selected processed data
-
-        Parameters
-        ----------
-        study : Study
-            The current study object
-        user : User
-            The current user object
-        callback : function
-            The callback function to call with the results once the processing
-            is done
-        """
-        pd_id = int(self.get_argument('processed_data_id'))
-
-        try:
-            ProcessedData.delete(pd_id)
-            msg = ("Processed data %d has been deleted" % pd_id)
-            msg_level = "success"
-            pd_id = None
-        except Exception as e:
-            msg = ("Couldn't remove processed data %d: %s" %
-                   (pd_id, str(e)))
-            msg_level = "danger"
-
-        callback((msg, msg_level, 'processed_data_tab', pd_id, None))
+class StudyIndexHandler(BaseHandler):
+    def study_prep_proxy(self, study):
+        """Proxies expected json from the API for existing prep templates"""
+        prep_info = {}
+        for dtype in study.data_types:
+            prep_info[dtype] = []
+            for prep in study.prep_templates(dtype):
+                start_artifact = prep.artifact
+                info = {
+                    'name': 'PREP %d NAME' % prep.id,
+                    'id': prep.id,
+                    'status': prep.status,
+                    'start_artifact': start_artifact.artifact_type,
+                    'start_artifact_id': start_artifact.id,
+                    'last_artifact': 'TODO'
+                }
+                prep_info[dtype].append(info)
+        return prep_info
+
+    def study_data_types_proxy(self):
+        """Proxies expected json from the API for available data types"""
+        data_types = Study.all_data_types()
+        return data_types if data_types else []
+
+    def study_info_proxy(self, study):
+        """Proxies expected json from the API for base study info"""
+        study_info = Study.get_info([study.id])[0]
+
+        pi = StudyPerson(study_info["principal_investigator_id"])
+        study_info["principal_investigator"] = '%s (%s)' % (
+            pi.name, pi.affiliation)
+        del study_info["principal_investigator_id"]
```

Comment:
Because just the ID comes from the database, not the actual studyperson object. Need to get the real info and replace the ID with the real info.